### PR TITLE
Added database and authentication

### PR DESCRIPTION
**Pull request by roberte777**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -2,66 +2,144 @@
...
+  name  String
+  phone String
+}
+
+model Shift {
+  id       Int                @id @default(autoincrement())
+  name     String
+  start    DateTime
+  end      DateTime
+  users    UserShiftMapping[]
```
Comment (TrevorAupperle) [inline prisma/schema.prisma:84]: Where is the `typeId`, `infoId`, `contactId`, `slots`,`day`, `notes`?

Comment (TrevorAupperle) [reply prisma/schema.prisma:84]: Also `name` field comes from `infoId`.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -2,66 +2,144 @@
...
+
+model UserShiftMapping {
+  id      Int    @id @default(autoincrement())
+  userId  String
+  shiftId Int
+  user    User   @relation(fields: [userId], references: [id])
+  shift   Shift  @relation(fields: [shiftId], references: [id])
+}
+
+model ShiftLog {
```
Comment (TrevorAupperle) [inline prisma/schema.prisma:51]: Need `description` and `type` fields.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -2,66 +2,144 @@
...
+  provider          String
+  providerAccountId String
+  refresh_token     String? @db.Text
+  access_token      String? @db.Text
+  expires_at        Int?
+  ext_expires_in    Int?
+  token_type        String?
+  scope             String?
+  id_token          String? @db.Text
+  session_state     String?
```
Comment (TrevorAupperle) [inline prisma/schema.prisma:102]: Camel case all of these.

File: prisma/schema.prisma [hunk 4]
```diff
@@ -2,66 +2,144 @@
...
+  name              String?            @db.Text
+  email             String?            @unique
+  phone             String?            @unique
+  year              Int                @default(-1)
+  major             String?
+  birthday          DateTime?
+  experience        Int                @default(0)
+  completedShifts   Int                @default(0)
+  shiftLeaderOfWeek Int                @default(0)
+  role_id           Int                @default(0)
```
Comment (TrevorAupperle) [inline prisma/schema.prisma:127]: Camel case this.

File: prisma/schema.prisma [hunk 5]
```diff
@@ -2,66 +2,144 @@
...
+  emailVerified     DateTime?
+  image             String?            @db.Text
+  accounts          Account[]
+  sessions          Session[]
+  Availability      Availability[]
+  Role              Role?              @relation(fields: [roleId], references: [id])
+  roleId            Int?
```
Comment (TrevorAupperle) [inline prisma/schema.prisma:134]: Duplicate `roleId`?

**Review by TrevorAupperle — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 3 above)
Comment (roberte777) [reply prisma/schema.prisma:102]: Those are next auth and don't get camel cased

File: prisma/schema.prisma (hunk 2 above)
Comment (roberte777) [reply prisma/schema.prisma:51]: what is the type field for again? What does it mean in real world terms

**Review by TrevorAupperle — APPROVED** [review]

**Pull request merged**